main.go

```diff
@@ -124,202 +120,142 @@ func main() {
 }
 
 func prepareCharts(c *cli.Context) {
-	repoRoot, err := os.Getwd()
-	if err != nil {
-		logrus.Fatalf("Unable to get current working directory: %s", err)
-	}
-	packages, err := charts.GetPackages(repoRoot, CurrentPackage)
-	if err != nil {
-		logrus.Fatal(err)
-	}
-	if len(packages) == 0 {
-		logrus.Fatalf("Could not find any packages in packages/")
-	}
+	packages := getPackages()
 	for _, p := range packages {
-		if err = p.Prepare(); err != nil {
+		if err := p.Prepare(); err != nil {
 			logrus.Fatal(err)
 		}
 	}
 }
 
 func generatePatch(c *cli.Context) {
-	repoRoot, err := os.Getwd()
-	if err != nil {
-		logrus.Fatalf("Unable to get current working directory: %s", err)
-	}
-	packages, err := charts.GetPackages(repoRoot, CurrentPackage)
-	if err != nil {
-		logrus.Fatal(err)
-	}
-	if len(packages) == 0 {
-		logrus.Fatalf("Could not find any packages in packages/")
-	}
+	packages := getPackages()
 	for _, p := range packages {
-		if err = p.GeneratePatch(); err != nil {
+		if err := p.GeneratePatch(); err != nil {
 			logrus.Fatal(err)
 		}
 	}
 }
 
 func generateCharts(c *cli.Context) {
-	repoRoot, err := os.Getwd()
-	if err != nil {
-		logrus.Fatalf("Unable to get current working directory: %s", err)
+	repo, _, status := getGitInfo()
+	if !status.IsClean() {
+		logrus.Warnf("Git is not clean:\n%s", status)
+		logrus.Fatal("Repository must be clean to generate charts")
 	}
-	packages, err := charts.GetPackages(repoRoot, CurrentPackage)
+	currentBranchRefName, err := repository.GetCurrentBranchRefName(repo)
 	if err != nil {
-		logrus.Fatal(err)
-	}
-	if len(packages) == 0 {
-		logrus.Fatalf("Could not find any packages in packages/")
+		logrus.Warnf("Due to limitations in the Git library used for the scripts, we cannot generate charts in a detached HEAD state.")
```

Comment:
Nit: looks like there is not parameter so logrus.Warn should be enough

Reply:
updated